test/cli-spec.js

```diff
@@ -188,4 +188,49 @@ describe('command line interface', function () {
       ) === false,
     );
   });
+  it('should unpack static framework with all underlying symlinks unpacked', async () => {
+    await execAsar(
+      'p test/input/packthis-with-symlink/ tmp/packthis-with-symlink.asar --unpack *.txt --unpack-dir "{dir2/subdir,Hello.framework,WindowsMklink}" --exclude-hidden',
+    );
+    assert.ok(fs.existsSync('tmp/packthis-with-symlink.asar.unpacked/A/real.txt'));
+    assert.ok(
+      fs.existsSync('tmp/packthis-with-symlink.asar.unpacked/Hello.framework/Versions/A/Hello'),
+    );
+    assert.ok(fs.existsSync('tmp/packthis-with-symlink.asar.unpacked/WindowsMklink/SymlinkedDir'));
+    assert.ok(fs.existsSync('tmp/packthis-with-symlink.asar.unpacked/WindowsMklink/Test/test.txt'));
+
+    // symlinks in Hello.framework can only be "followed" on mac/linux, so we have separate Windows-created symlinks via `mklink` as another test (WindowsMklink folder)
+    if (process.platform !== 'win32') {
+      assert.equal(
+        fs.readlinkSync('tmp/packthis-with-symlink.asar.unpacked/real.txt'),
+        'Current/real.txt',
+      );
+      assert.equal(
+        fs.readlinkSync('tmp/packthis-with-symlink.asar.unpacked/Hello.framework/Hello'),
+        'Versions/Current/Hello',
+      );
+      assert.ok(
+        fs
+          .realpathSync(
+            'tmp/packthis-with-symlink.asar.unpacked/Hello.framework/Versions/Current/Hello',
+          )
+          .endsWith('tmp/packthis-with-symlink.asar.unpacked/Hello.framework/Versions/A/Hello'),
+      );
+      assert.equal(
+        fs.readlinkSync('tmp/packthis-with-symlink.asar.unpacked/Hello.framework/Headers'),
+        'Versions/Current/Headers',
+      );
+      assert.ok(
+        fs
+          .realpathSync('tmp/packthis-with-symlink.asar.unpacked/Hello.framework/Headers')
+          .endsWith('tmp/packthis-with-symlink.asar.unpacked/Hello.framework/Versions/A/Headers'),
+      );
+    } else {
+      assert.ok(
```

Comment:
After git cloning on Windows, SymlinkedDir is not a symlink but rather a regular file. I suggest only testing this on macOS/Linux. No need to test on Windows since you're not handling symlink in window.

https://github.com/mmaietta/asar/blob/1676a0ea070967f818167164be5bf7d0e378fb41/src/disk.ts#L65-L68